File: week_2/valid_anagram.py
def are_anagrams(first_string, second_string):
    first_word_hash_table = {}
    second_word_hash_table = {}

    for char in first_string:
        if char in first_word_hash_table:
            first_word_hash_table[char] += 1
        else:
            first_word_hash_table[char] = 1

    for char in second_string:
        if char in second_word_hash_table:
            second_word_hash_table[char] += 1
        else:
            second_word_hash_table[char] = 1

    return first_word_hash_table == second_word_hash_table

File: week_2/valid_anagram.js

const objectsEqual = (o1, o2) => Object.keys(o1).length
=== Object.keys(o2).length && Object.keys(o1).every((p) => o1[p] === o2[p]);

function areAnagrams(firstString, secondString) { 
    const firstWordHashTable = {};
    const secondWordHashTable = {};

    for (const char of firstString) { 
        if (firstWordHashTable[char]) {
            firstWordHashTable[char] += 1; 
        } else {
            firstWordHashTable[char] = 1;
        }
    }

    for (const char of secondString) {
        if (secondWordHashTable[char]) {
            secondWordHashTable[char] += 1;
        } else {
            secondWordHashTable[char] = 1;
        }
    }

    return objectsEqual(firstWordHashTable, secondWordHashTable);
}
